### PR TITLE
Add client for idris2-lsp

### PR DESCRIPTION
**Pull request by skykanin**
# Idris support
Register client for [idris2-lsp](https://github.com/idris-community/idris2-lsp)

### CODE REVIEW

File: clients/lsp-idris.el [hunk 1]
```diff
@@ -0,0 +1,73 @@
+;;; lsp-idris.el --- Description -*- lexical-binding: t; -*-
+;;
+;; Copyright (C) 2021 skykanin
```
Comment (yyoncho) [inline clients/lsp-idris.el:3]: 2022

File: clients/lsp-idris.el [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+(require 'lsp-mode)
+(require 'lsp-semantic-tokens)
+
+(defgroup lsp-idris nil
+  "LSP support for Idris."
+  :link '(url-link "https://github.com/idris-community/idris2-lsp")
+  :group 'lsp-mode
+  :tag "Lsp Idirs"
+  :package-version '(lsp-mode . "8.0.0"))
```
Comment (yyoncho) [inline clients/lsp-idris.el:37]: can you change these to 8.0.1 in the whole file?

File: clients/lsp-idris.el [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  (make-lsp-client
+   :new-connection (lsp-stdio-connection "idris2-lsp")
+   ;; Activate lsp on idris or idris2 buffers
+   :activation-fn (lsp-activate-on "idris" "idris2")
+   ;; This should run under idris-mode and idris2-mode.
+   :major-modes modes
+   :language-id "idris"
+   :server-id 'idris2-lsp))
+
+;; Add lsp hook for every idris major mode
```
Comment (yyoncho) [inline clients/lsp-idris.el:69]: should be deleted

File: clients/lsp-idris.el [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  "Traces the communication between VS Code and the language server."
+  :group 'lsp-idris
+  :type '(choice (:tag "off" "messages" "verbose"))
+  :package-version '(lsp-mode . "8.0.0"))
+
+(lsp-register-custom-settings
+  '(("idris2-lsp.trace.server" lsp-idris2-lsp-trace-server)
+    ("idris2-lsp.path" lsp-idris2-lsp-path)))
+
+;; Major modes for Idris
```
Comment (yyoncho) [inline clients/lsp-idris.el:55]: should be deleted

File: clients/lsp-idris.el [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+(defconst modes '(idris-mode idris2-mode))
+
+;; Register the client itself
+(lsp-register-client
+  (make-lsp-client
+   :new-connection (lsp-stdio-connection "idris2-lsp")
```
Comment (yyoncho) [inline clients/lsp-idris.el:61]: use `lsp-idris2-lsp-path` instead of hardcoding it.

File: clients/lsp-idris.el [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  (make-lsp-client
+   :new-connection (lsp-stdio-connection lsp-idris2-lsp-path)
+   ;; Activate lsp on idris or idris2 buffers
+   :activation-fn (lsp-activate-on "idris" "idris2")
+   ;; This should run under idris-mode and idris2-mode.
+   :major-modes idris-modes
+   :language-id "idris"
+   :server-id 'idris2-lsp))
+
+(mapc (lambda (mode) (add-hook mode #'lsp!)) idris-modes)
```
Comment (yyoncho) [inline clients/lsp-idris.el:68]: remove this

File: clients/lsp-idris.el [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  "Traces the communication between VS Code and the language server."
+  :group 'lsp-idris
+  :type '(choice (:tag "off" "messages" "verbose"))
+  :package-version '(lsp-mode . "8.0.1"))
+
+(lsp-register-custom-settings
+  '(("idris2-lsp.trace.server" lsp-idris2-lsp-trace-server)
+    ("idris2-lsp.path" lsp-idris2-lsp-path)))
+
+(defconst idris-modes '(idris-mode idris2-mode))
```
Comment (yyoncho) [inline clients/lsp-idris.el:55]: remove

File: lsp-mode.el [hunk 8]
```diff
@@ -4893,10 +4895,32 @@ If INCLUDE-DECLARATION is non-nil, request the server to include declarations."
    (->> lsp--cur-workspace lsp--workspace-client lsp--client-response-handlers (remhash id))
    (lsp-notify "$/cancelRequest" `(:id ,id))))
 
-(defun lsp-eldoc-function ()
```
Comment (yyoncho) [inline lsp-mode.el:4896]: revert, this is probably a bad merge.

File: clients/lsp-idris.el (hunk 6 above)
Comment (skykanin) [reply clients/lsp-idris.el:68]: Is this something the user should configure or why do I need to remove this?

Comment (yyoncho) [reply clients/lsp-idris.el:68]: Yes.

**Comment by yyoncho** [comment]
Thank you!

**Pull request merged**